core/src/main/java/feast/core/model/FeatureTable.java

```diff
@@ -279,6 +311,11 @@ public boolean hasAllLabels(Map<String, String> labelsFilter) {
     return TypeConversion.convertJsonStringToMap(getLabelsJSON());
   }
 
+  public void delete() {
+    this.setDeleted(true);
+    this.setRevision(0);
```

Comment:
how revision is being used?

Reply:
Revision is initialized whenever a feature table is created. As our deletion keeps the entry in the database, we should reset it to 0 since future apply to this deleted feature table would yield a new one.